dwave/gate/circuit.py

```diff
@@ -220,6 +230,57 @@ def bits(self) -> ClassicalRegister:
 
         return bit_reg
 
+    @property
+    def state(self) -> Optional[NDArray]:
+        """The resulting state after simulating the circuit."""
+        if self._state is None and self._density_matrix is not None:
+            warnings.warn("State is mixed. Use 'Circuit.density_matrix' to access.")
```

Comment:
Also, why a warning rather than an error?

Reply:
Ah, I find it nicer not to raise an error if there is no state stored. If there's no state at all, it will just return `None`. In this case it will just raise a warning if the state only exists as a density matrix to notify the user that there is infact a state stored, but only as a density matrix (a mixed one).

Reply:
I guess I can think of two uses I might have as a user
a) I want to check whether the state is in the density matrix as part of my flow. In which case I want to do things like
```
if circuit.state is None:
    # do something else
```
in that case I don't think I want a warning.
b) It's always a mistake to access it in that state, in which case I would want an error.